renderer/brave_content_renderer_client.cc

```diff
@@ -45,6 +49,31 @@
 #include "brave/components/playlist/renderer/playlist_render_frame_observer.h"
 #endif
 
+#if BUILDFLAG(ENABLE_WIDEVINE)
+#include "media/base/key_system_info.h"
+#include "third_party/widevine/cdm/widevine_cdm_common.h"
+#endif
+
+namespace {
+void MaybeRemoveWidevineSupport(media::GetSupportedKeySystemsCB cb,
+                                media::KeySystemInfos key_systems) {
+#if BUILDFLAG(ENABLE_WIDEVINE) && BUILDFLAG(IS_ANDROID)
```

Comment:
I don't think this actually needs an `IS_ANDROID` guard. What do we do on desktop when this is disabled? Uninstall widevine? Maybe we can simplify that and just use this for both?

Reply:
If I'm not mistaken, on desktop because Widevine is played through component, if the user hasn't installed it will prompt, or if the user declined and chose 'do not show' then it just does nothing and because there is no component widevine just won't play. There is no specific action for the disabled state.

Reply:
this actually looks broken on desktop because the widevine opt-in is per-profile, but widevine is installed globally

Reply:
when disabling on desktop it uninstalls the component, but that's global and the pref is per-profile

Reply:
@bridiver `kWidevineOptedIn` is local state prefs now.
It was per-profile prefs but migrated to local state.
I assume you saw `widevine_opted_in` in brave://prefs-internals.
but it's deprecated and registered for migration.
It could be deleted safely now as migration logic was added long times ago(11/2019).

Reply:
It doesn't hurt to add this logic to desktop though, shouldn't have any adversarial effect I believe?